### PR TITLE
Isochrone: add marker on isochrone center

### PR DESCRIPTION
**Pull request by ZoeBrunet**

### CODE REVIEW

File: js/map.js [hunk 1]
```diff
@@ -80,8 +80,23 @@ var map = {
         },
         isochrone: function(context, json) {
             if (! ('geojson' in json)) { return []; }
+            var draw_section_option = map.DrawSectionOption.DRAWNEITHER;
+            draw_section_option |= map.DrawSectionOption.DRAWSTART;
             var color = context.min_duration_color[json.min_duration];
-            return map._makePolygon(context, 'isochrone', json.geojson, json, color);
+            console.log(color);
```
Comment (xlqian) [inline js/map.js]: debug info?

File: js/map.js [hunk 2]
```diff
@@ -80,8 +80,23 @@ var map = {
         },
         isochrone: function(context, json) {
             if (! ('geojson' in json)) { return []; }
+            var draw_section_option = map.DrawSectionOption.DRAWNEITHER;
+            draw_section_option |= map.DrawSectionOption.DRAWSTART;
```
Comment (xlqian) [inline js/map.js]: `var draw_section_option = map.DrawSectionOption.DRAWNEITHER | map.DrawSectionOption.DRAWSTART`?

btw, `var draw_section_option` is not used... :)

File: js/map.js [hunk 3]
```diff
@@ -80,8 +80,23 @@ var map = {
...
             var color = context.min_duration_color[json.min_duration];
-            return map._makePolygon(context, 'isochrone', json.geojson, json, color);
+            console.log(color);
+            var place;
+            if(('from' in json)) {
```
Comment (xlqian) [inline js/map.js]: hmmmm, i think `if('from' in json)` should be enough

File: js/map.js [hunk 4]
```diff
@@ -80,8 +80,23 @@ var map = {
...
+                place = json.from;
+            }
+            if(('to' in json)) {
+                place = json.to;
+            }
```
Comment (xlqian) [inline js/map.js]: so, if both `from` and `to` exist in json, `to` will override `from`?

File: js/map.js [hunk 5]
```diff
@@ -80,8 +80,23 @@ var map = {
...
+            if (color.color == '00ff00') {
```
Comment (xlqian) [inline js/map.js]: in this specific case, hard-coded string is evil. If we change the "starting color", there'll not be marker

File: js/map.js (hunk 4 above)
Comment (ZoeBrunet) [inline js/map.js]: They cannot both exist

File: js/map.js [hunk 6]
```diff
@@ -244,17 +253,16 @@ var map = {
             // when section is Walking
             var from = json.from;
             var to = json.to;
-            if (! from || ! to) { return markers; }
             var label_from = null;
             var label_to = null;
-            if (map._should_draw_section_start(draw_section_option)) {
+            if (from && map._should_draw_section_start(draw_section_option)) {
                 label_from = map.STARTTEXT;
+                markers = markers.concat(map._makeMarker(context, 'place', from, color, true, label_from));
```
Comment (TeXitoi) [inline js/map.js]: you can `markers.push(...)`

File: js/map.js [hunk 7]
```diff
@@ -244,17 +253,16 @@ var map = {
...
             }
-            if (map._should_draw_section_end(draw_section_option)) {
+            if (to && map._should_draw_section_end(draw_section_option)) {
                 label_to = map.ENDTEXT;
+                markers = markers.concat(map._makeMarker(context, 'place', to, color, true, label_to));
```
Comment (TeXitoi) [inline js/map.js]: same

File: js/map.js [hunk 8]
```diff
@@ -81,7 +81,16 @@ var map = {
         isochrone: function(context, json) {
             if (! ('geojson' in json)) { return []; }
             var color = context.min_duration_color[json.min_duration];
-            return map._makePolygon(context, 'isochrone', json.geojson, json, color);
+            var draw_section_option = map.DrawSectionOption.DRAWNEITHER;
```
Comment (TeXitoi) [inline js/map.js]: I think you can just `DRAWBOTH` and `_makeStopTimesMarker` will do the job filtering if `from` or `to` is here.

File: js/map.js [hunk 9]
```diff
@@ -81,7 +81,16 @@ var map = {
...
+            color_marker = '#000000'
+            if('from' in json) {
+                draw_section_option |= map.DrawSectionOption.DRAWSTART;
+            }
+            if('to' in json) {
+                draw_section_option |= map.DrawSectionOption.DRAWEND;
+            }
```
Comment (TeXitoi) [inline js/map.js]: Then, you can remove these 2 if.

File: js/map.js [hunk 10]
```diff
@@ -257,11 +251,11 @@ var map = {
             var label_to = null;
             if (from && map._should_draw_section_start(draw_section_option)) {
                 label_from = map.STARTTEXT;
-                markers = markers.concat(map._makeMarker(context, 'place', from, color, true, label_from));
+                markers.push(map._makeMarker(context, 'place', from, color, true, label_from)[0]);
             }
             if (to && map._should_draw_section_end(draw_section_option)) {
                 label_to = map.ENDTEXT;
-                markers = markers.concat(map._makeMarker(context, 'place', to, color, true, label_to));
+                 markers.push(map._makeMarker(context, 'place', to, color, true, label_to)[0]);
```
Comment (TeXitoi) [inline js/map.js]: bad indent.

File: js/map.js [hunk 11]
```diff
@@ -244,17 +247,16 @@ var map = {
...
             var label_to = null;
-            if (map._should_draw_section_start(draw_section_option)) {
+            if (from && map._should_draw_section_start(draw_section_option)) {
                 label_from = map.STARTTEXT;
+                markers.push(map._makeMarker(context, 'place', from, color, true, label_from)[0]);
             }
-            if (map._should_draw_section_end(draw_section_option)) {
+            if (to && map._should_draw_section_end(draw_section_option)) {
                 label_to = map.ENDTEXT;
+                 markers.push(map._makeMarker(context, 'place', to, color, true, label_to)[0]);
```
Comment (TeXitoi) [inline js/map.js]: bad indent

**Pull request merged**